# Source: tonistiigi/binfmt patches/buildkit-direct-execve-v8.1/0006-linux-user-use-GLib-to-remember-the-program-name.patch
#   https://raw.githubusercontent.com/tonistiigi/binfmt/master/patches/buildkit-direct-execve-v8.1/0006-linux-user-use-GLib-to-remember-the-program-name.patch
# Commit: e4620e68 ("update qemu to 8.1.4", 2023-12-01); tree 4b16346a86506f3c8e298aba5f5972d9446d8ddf
# Applies to: qemu 8.2.2 (see QEMU_VERSION in build-qemu.sh)

From 840dd52102da8ba46a8d4d55ceae099a98fdd137 Mon Sep 17 00:00:00 2001
From: CrazyMax <crazy-max@users.noreply.github.com>
Date: Wed, 3 May 2023 20:57:04 +0200
Subject: [PATCH] linux-user: use GLib to remember the program name

Signed-off-by: CrazyMax <crazy-max@users.noreply.github.com>
---
 linux-user/syscall.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/linux-user/syscall.c b/linux-user/syscall.c
index 0ce9f207be..173d56ba5e 100644
--- a/linux-user/syscall.c
+++ b/linux-user/syscall.c
@@ -8447,7 +8447,7 @@ static int do_execveat(CPUArchState *cpu_env, int dirfd,
      * where qemu_progname is the error message prefix for qemu.
      * Note: if pathname is relative, it will be prepended with the current working directory.
     */
-    argp[0] = (char*)error_get_progname();
+    argp[0] = (char*)g_get_prgname();
     argp[1] = (char*)"-0";
     argp[2] = (char*)lock_user_string(addr);
     argp[3] = (char*)prepend_workdir_if_relative(p);
-- 
2.34.0

